code/modules/mob/living/carbon/human/death.dm

```diff
@@ -38,6 +38,12 @@
 	if(mind && species && (species.name == "Manifested") && (mind in ticker.mode.cult))//manifested ghosts are removed from the cult once their bodies are destroyed
 		ticker.mode.update_cult_icons_removed(mind)
 		ticker.mode.cult -= mind
+
+	for(var/obj/Overlays/O in obj_overlays)
+		returnToPool(O)
+	
+	obj_overlays = null
```

Comment:
As long as the overlays list gets re-initialized somewhere when it's return from pool okay